tests/test_config_parser.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import torch
+from parameterized import parameterized
+
+from monai.apps import ConfigParser
+from monai.transforms import LoadImaged
+from tests.utils import skip_if_windows
+
+TEST_CASE_1 = [
+    dict(pkgs=["monai"], modules=["transforms"]),
+    {"name": "LoadImaged", "args": {"keys": ["image"]}},
+    LoadImaged,
+]
+# test python `path`
+TEST_CASE_2 = [
+    dict(pkgs=[], modules=[]),
+    {"path": "monai.transforms.LoadImaged", "args": {"keys": ["image"]}},
+    LoadImaged,
+]
+# test `disabled`
+TEST_CASE_3 = [
+    dict(pkgs=["monai"], modules=["transforms"]),
+    {"name": "LoadImaged", "disabled": True, "args": {"keys": ["image"]}},
+    None,
+]
+# test non-monai modules
+TEST_CASE_4 = [
+    dict(pkgs=["torch.optim", "monai"], modules=["adam"]),
+    {"name": "Adam", "args": {"params": torch.nn.PReLU().parameters(), "lr": 1e-4}},
+    torch.optim.Adam,
+]
+
+
+class TestConfigParser(unittest.TestCase):
+    @parameterized.expand([TEST_CASE_1, TEST_CASE_2, TEST_CASE_3])
+    def test_type(self, input_param, test_input, output_type):
+        configer = ConfigParser(**input_param)
+        result = configer.build_component(test_input)
+        if result is not None:
+            self.assertTrue(isinstance(result, output_type))
+            self.assertEqual(result.keys[0], "image")
+        else:
+            # test `disabled` works fine
+            self.assertEqual(result, output_type)
+
+
+@skip_if_windows
```

Comment:
How come this doesn't work on Windows?

Reply:
Hi @rijobro ,

This is an error when parsing the paths of PyTorch:
```
ERROR: test_type_3 (tests.test_config_parser.TestConfigParser)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\parameterized\parameterized.py", line 533, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "D:\a\MONAI\MONAI\tests\test_config_parser.py", line 47, in test_type
    configer = ConfigParser(**input_param)
  File "D:\a\MONAI\MONAI\monai\apps\mmars\config_parser.py", line 68, in __init__
    self.module_scanner = ModuleScanner(pkgs=pkgs, modules=modules)
  File "D:\a\MONAI\MONAI\monai\apps\mmars\config_parser.py", line 34, in __init__
    self._class_table = self._create_classes_table()
  File "D:\a\MONAI\MONAI\monai\apps\mmars\config_parser.py", line 41, in _create_classes_table
    for _, modname, _ in pkgutil.walk_packages(path=package.__path__, prefix=package.__name__ + "."):
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\pkgutil.py", line 107, in walk_packages
    yield from walk_packages(path, info.name+'.', onerror)
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\pkgutil.py", line 92, in walk_packages
    __import__(info.name)
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\torch\distributed\_sharded_tensor\__init__.py", line 15, in <module>
    from .api import (
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\torch\distributed\_sharded_tensor\api.py", line 145, in <module>
    def _register_remote_shards(sharded_tensor_id: int, rrefs: List[rpc.RRef[Shard]], rpc_rank: int):
AttributeError: module 'torch.distributed.rpc' has no attribute 'RRef'
```
I will check it deeper.

Thanks.

Reply:
Hi @rijobro ,

The issue had been fixed in the latest commit.

Thanks.

Reply:
Great, thanks!